beakerx/beakerx/static/tree.js

```diff
@@ -264,107 +276,125 @@ define(function (require) {
-          payload['jvm_options'] = {};
-          var values = [];
-          var other_property = $('#other_property input');
-          other_property.each(function () {
-            var value = $(this).val().trim();
-            if (value.length > 0) {
-              values.push(value);
-            }
-          });
-          payload['jvm_options']['other'] = values;
-          var java_values = {};
-          var java_property = $('#properties_property div');
-          java_property.each(function () {
-            var children = $($(this).children());
-            var value = $(children.get(1)).val().trim();
-            var name = $(children.get(0)).val().trim();
-
-            if (name.length > 0) {
-              java_values[name] = value
-            }
+        if (window.location.hash === '#beakerx') {
+          $('#beakerx_tab').tab('show');
+          version.load();
+        }
+      }
+    });
+  }
 
-          });
-          var default_property = $('#default_options input');
-          default_property.each(function () {
-            var value = $(this).val().trim();
-            if (value.length > 0) {
-              payload['jvm_options']['heap_GB'] = value
-            }
-          });
-          payload['jvm_options']['properties'] = java_values;
-          settings.setVariables(JSON.stringify({'beakerx': payload}));
+  function _createBeakerxTab(tab_html) {
+    $(".tab-content").append($(tab_html));
+    $("#tabs").append(
+      $('<li>').append(
+        $('<a>', {
+          id: 'beakerx_tab',
+          href: '#beakerx',
+          'data-toggle': 'tab',
+          text: 'BeakerX'
+        }).click(function (e) {
+          if (window.location.hash === '#beakerx') {
+            return;
+          }
+
+          window.history.pushState(null, null, '#beakerx');
+          version.load();
           settings.load();
-        });
-
-        $("#add_property_jvm_sett").click(function (event) {
-          event.preventDefault();
-          var parent = $('#properties_property');
-
-          var opts = {
-            name: "",
-            value: "",
-            parent: parent,
-            add_label: true
-          };
-
-          settings.appendField(opts);
-        });
-
-        $("#add_option_jvm_sett").click(function (event) {
-          event.preventDefault();
-          var fieldset = $('#other_property');
-          settings.createField(fieldset);
-        });
-        $('#heap_GB').keyup(InpuChanged);
-        $(".tab-content").submit();
-
-        if (window.location.hash === '#beakerx') {
-          $('#beakerx_tab').click();
+        })
+      )
+    );
+  }
+  function _setupFormEvents() {
+    $("#add_property_jvm_sett").click(_onJvmPropertyAddClickedHandler);
+    $("#add_option_jvm_sett").click(_onJvmOptionAddClickedHandler);
+    $('#heap_GB').keyup(InputChanged);
+  }
+  function _setupContinuousSync() {
+    var lastSyncedTs = null;
+    function synchronizeTick() {
+      setTimeout(function() {
```

Comment:
wouldn't it be better to use _.debounce instead of implementing the same idea ourselves?

Reply:
@scottdraves sorry i missed this comment